src/vivarium/framework/results/manager.py

```diff
@@ -71,36 +70,36 @@ def setup(self, builder: "Builder"):
 
         builder.value.register_value_modifier("metrics", self.get_results)
 
-    def on_post_setup(self, event: Event):
-        # update self._metrics to have all output keys
-        def create_measure_specific_keys(measure: str, stratifications: List[str]) -> None:
-            measure_str = f"MEASURE_{measure}"
-            individual_stratification_strings = [
-                [
-                    f"{stratification.name.upper()}_{category}"
-                    for category in stratification.categories
-                ]
-                for stratification in sorted(
-                    self._results_context.stratifications, key=lambda x: x.name
-                )
-                if stratification.name in stratifications
-            ]
-            for complete_stratifications in itertools.product(
-                *individual_stratification_strings
-            ):
-                key = (
-                    measure_str + "_" + "_".join(complete_stratifications)
-                    if complete_stratifications
-                    else measure_str
-                )
-                self._metrics[key] = 0
-
-        for event in self._results_context.observations:
-            for (_, stratifications), observations in self._results_context.observations[
-                event
-            ].items():
+    def on_post_setup(self, _: Event):
+        """Initialize self._metrics with 0s dataframe for each measure and all stratifications"""
+        for event_name in self._results_context.observations:
+            for (
+                _pop_filter,
+                stratification_names,
+            ), observations in self._results_context.observations[event_name].items():
                 for measure, *_ in observations:
-                    create_measure_specific_keys(measure, stratifications)
+                    stratification_values = []
+                    for stratification in [
+                        s
+                        for s in self._results_context.stratifications
+                        if s.name in stratification_names
+                    ]:
+                        stratification_values.append(
+                            (stratification.name, stratification.categories)
+                        )
```

Comment:
Nit: I think this is easier to read:
```suggestion
                    observation_stratifications = [
                        stratification
                        for stratification in self._results_context.stratifications
                        if stratification.name in stratification_names
                    ]
                    stratification_values = [
                        (stratification.name, stratification.categories)
                        for stratification in observer_stratifications
                    ]
```

Reply:
I also think `stratification_values` makes more sense to be a dictionary, rather than a list of tuples.